tiny_dnn/core/framework/tensor.h

```diff
@@ -231,150 +189,91 @@ class Tensor {
     return *this;
   }
 
-  // TODO(Randl): variadic template version of reshape
-  // TODO(Randl): non-checked version
-  void reshape(const std::array<size_t, kDimensions> &sz) {
-    static_assert(!kConst, "Non-constant operation on constant Tensor");
-    // No size change for reshape
-    if (calcSize() != product(sz)) {
-      throw nn_error("Reshape to Tensor of different size.");
-    }
-    shape_ = sz;
-  }
-
-  void resize(const std::array<size_t, kDimensions> &sz) {
-    if (offset_ != 0 || size_ != storage_ptr_->size()) {
-      throw nn_error("Resize of partial view is impossible.");
-    }
-    shape_ = sz;
-    storage_ptr_->resize(std::vector<size_t>(sz.begin(), sz.end()));
-  }
-
-  size_t size() const { return size_; }
-
-  Tensor operator[](size_t index) {
-    return Tensor(
-      storage_ptr_, offset_ + index * size_ / shape_[0],
-      std::array<size_t, kDimensions - 1>(shape_.begin() + 1, shape_.end()));
-  }
-
+  // TODO(Randl): checked version
   /**
-   * @brief Returns a sub view from the current tensor with a given size.
-   * The new tensor will share data with its parent tensor so that each time
-   * that data is modified, it will be updated in both directions.
-   *
-   * The new sub view tensor will be extracted assuming continuous data.
-   * The offset to shared data is assumed to be 0.
-   *
-   * @param new_shape The size for the new tensor
-   * @return An instance to the new tensor
-   *
-   * Usage:
-   *
-   *  Tensor<float_t, 4> t({2,2,2,2});            // we create a 4D tensor
-   *  Tensor<float_t, 4> t_view = t.view({2,2});  // we create a 2x2 matrix
-   * view
-   * with offset zero
-   *
+   * Reshape tensor
+   * @param shape new shape
    */
-  Tensor subView(std::initializer_list<size_t> const &new_shape) {
-    return subview_impl({}, new_shape);
-  }
+  void reshape(const std::vector<size_t> &shape) { storage_.reshape(shape); }
 
   /**
-   * @brief Returns a sub view from the current tensor with a given size.
-   * The new tensor will share data with its parent tensor so that each time
-   * that data is modified, it will be updated in both directions.
-   *
-   * The new sub view tensor will be extracted assuming continuous data.
-   *
-   * @param start The offset from the parent tensor
-   * @param new_shape The size for the new tensor
-   * @return An instance to the new tensor
-   *
-   * Usage:
-   *
-   *  Tensor<float_t, 4> t({2,2,2,2});                   // we create a 4D
-   * tensor
-   *  Tensor<float_t, 4> t_view = t.view({2,2}, {2,2});  // we create a 2x2
-   * matrix view from
-   *                                                     // offset 4.
+   * Reshape tensor
+   * @param shape new shape
    */
-  Tensor subView(std::initializer_list<size_t> const &start,
-                 std::initializer_list<size_t> const &new_shape) {
-    return subview_impl(start, new_shape);
-  }
+  void resize(const std::vector<size_t> &shape) { storage_.reshape(shape); }
```

Comment:
> if you mean absolute indexes than no
@Randl I checked the code.
https://github.com/QuantStack/xtensor/blob/master/include/xtensor/xcontainer.hpp#L789
The method doesn't rearrange elements which is against the intuition that the users can access the same elements with same indexes after calling the `resize` method.